### PR TITLE
use `bb.ensureInstance` instead of `bb.join` in renderer.ts

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
I haven't dug in, but it looks like the test failures here are related to your changes?

File: packages/@ember/-internals/glimmer/lib/renderer.ts [hunk 1]
```diff
@@ -87,7 +86,6 @@ class RootState {
     this.env = env;
     this.root = root;
     this.result = undefined;
-    this.shouldReflush = false;
```
Comment (bekzod) [inline packages/@ember/-internals/glimmer/lib/renderer.ts]: leftover from the past, so removed it

File: packages/@ember/-internals/glimmer/lib/renderer.ts [hunk 2]
```diff
@@ -68,7 +68,6 @@ class RootState {
   public env: Environment;
   public root: Component | OutletView;
   public result: RenderResult | undefined;
-  public shouldReflush: boolean;
```
Comment (bekzod) [inline packages/@ember/-internals/glimmer/lib/renderer.ts]: leftover from the past, so removed it

File: packages/@ember/-internals/glimmer/tests/integration/syntax/each-test.js [hunk 3]
```diff
@@ -555,7 +555,7 @@ class EachTest extends AbstractEachTest {
 
     runTask(() => set(this.objectAt(0), 'value', 3));
 
-    this.assertText('PrevNextPrev2NextPrev3Next');
```
Comment (bekzod) [inline packages/@ember/-internals/glimmer/tests/integration/syntax/each-test.js]: surprisingly this didn't assert before

**Comment by bekzod** [comment]
after checking test failures, I had to add `renderSettled` few places, probably because there is no extra run loop spawn there is not enough time to settle for those case, haven't had time to investigate too deep on that

**Comment by locks** [comment]
@bekzod you have some conflicts!

**Pull request closed without merging**